containerd-http-proxy/configure_http_proxy.yaml

```diff
@@ -0,0 +1,107 @@
+kind: DaemonSet
+apiVersion: apps/v1
+metadata:
+  name: containerd-http-proxy
+spec:
+  selector:
+    matchLabels:
+      name: containerd-http-proxy
+  template:
+    metadata:
+      labels:
+        name: containerd-http-proxy
+    spec:
+      hostPID: true
+      containers:
+      - name: startup-script
+        image: gke.gcr.io/debian-base:bookworm-v1.0.0-gke.1
+        imagePullPolicy: IfNotPresent
+        securityContext:
+          privileged: true
+        command:
+        - /bin/sh
+        - -c
+        - |
+          set -e
+          set -u
+
+          validate_proxy() {
+            input_string=$1
+
+            if echo "$input_string" | grep -qvE '^(HTTP_PROXY|HTTPS_PROXY|NO_PROXY)='; then
+                echo "Error: Input does not start with a valid prefix (HTTP_PROXY, HTTPS_PROXY, NO_PROXY). Input: '$input_string'"
+                exit 1
+            fi
+
+            if echo "$input_string" | grep -q ' '; then
+                echo "Error: Input cannot contain spaces. Input: '$input_string'"
+                exit 1
+            fi
+
+            if echo "$input_string" | sed 1d | grep -q .; then
+                echo "Error: Input cannot contain newline. Input: '$input_string'"
+                exit 1
+            fi
+          }
+
+          validate_proxy "${HTTP_PROXY:-}"
+          validate_proxy "${HTTPS_PROXY:-}"
+          validate_proxy "${NO_PROXY:-NO_PROXY=localhost}"
+
+          STARTUP_SCRIPT=$(cat <<END
+          set -e
+          set -u
+          mkdir -p /etc/systemd/system/containerd.service.d
+          echo "[Service]
+          Environment=${HTTP_PROXY}
+          Environment=${HTTPS_PROXY}
+          Environment=${NO_PROXY:-NO_PROXY=localhost}" > /etc/systemd/system/containerd.service.d/http-proxy.conf
+          echo "Reloading systemd management configuration"
+          systemctl daemon-reload
+          echo "Restarting containerd..."
+          systemctl restart containerd
+          END
+          )
+
+          CONFIG_PATH="/etc/systemd/system/containerd.service.d/http-proxy.conf"
+          CHECK_INTERVAL_SECONDS="30"
+          EXEC="nsenter -t 1 -m -u -i -n -p --"
```

Comment:
Running systemctl or restart containerd would need it. Mounting all the required directories and files is impractical.

Reply:
btw, this is what we already have in this repo: https://github.com/GoogleCloudPlatform/k8s-node-tools/blob/6f5f17af0971c54ddff77b000abda3788cba0965/container-insecure-registry/insecure-registry-config.yaml#L46

Would using the same image be better instead of nsenter?

Reply:
In other examples we do this: https://github.com/GoogleCloudPlatform/k8s-node-tools/blob/6f5f17af0971c54ddff77b000abda3788cba0965/configure-hugepages/configure-hugepages.yaml#L103

Reply:
not sure what chroot /host does and why we don't do it here

Reply:
The [implementation](https://github.com/kubernetes-retired/contrib/blob/master/startup-script/manage-startup-script.sh) of gcr.io/google-containers/startup-script is 8 years old, but still looks ok. @SergeyKanzhelev Is there any recommended way? Or it is case by case?

Reply:
do we need chroot /host for nsenter? Do you know why it is used in other Daemonsets in this repo?